needy/library.py

```diff
@@ -51,7 +51,7 @@ def __init__(self, needy, name, target=None, configuration=None, development_mod
         self.__configuration = configuration
         self.__directory = os.path.join(needy.needs_directory(), name)
         self.__development_mode = development_mode
-        self.__build_cache = BuildCache(self.needy.cache()) if self.needy.cache() else None
+        self.__build_caches = [BuildCache(c) for c in self.needy.needy_configuration().caches()]
```

Comment:
So all caches are build caches? If "build cache" and "cache" are synonymous, maybe we should pick one term and go with it. For future-proofing, I think I'd prefer to refer to them as "build caches" everywhere.